src/main/java/net/mcreator/gradle/GradleUtils.java

```diff
@@ -80,6 +82,12 @@ public static String getJavaHome() {
 				LOG.error("Java home override from preferences is not valid!");
 		}
 
+		// if the generator has a JDK, we set JAVA_HOME to generator's JDK
+		if (UserFolderManager.getSpecificJDK(genConfig.getJDKVersion() + "/bin/javac.exe").isFile() || UserFolderManager

```

Comment:
This step should be skipped if the generator configuration does not specify jdk. Also refactor genConfig to full name